packages/global-search-module/src/settings.vue

```diff
@@ -0,0 +1,178 @@
+<script setup lang="ts">
+import { useStores } from '@directus/extensions-sdk';
+import { clone, merge } from 'lodash-es';
+import { ref, computed, watch, type Ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useMagicKeys } from '@vueuse/core';
+import { ZodError } from 'zod';
+import type { ValidationError } from '@directus/types';
+import { useIsAdmin } from './composables/use-is-admin';
+
+import { fields } from './settings-fields';
+import SearchNavigation from './components/navigation.vue';
+import { SearchConfig, defaultSettings } from './types';
+import { zodErrorToValidationErrors } from './utils/errors';
+
+const { t } = useI18n();
+const { useSettingsStore, useFieldsStore } = useStores();
+
+const fieldsStore = useFieldsStore();
+const settingsStore = useSettingsStore();
+
+const isAdmin = useIsAdmin();
+
+const globalSearchSettings = computed(() => settingsStore.settings.global_search_settings);
+const globalSearchSettingsField = fieldsStore.getField('directus_settings', 'global_search_settings');
+
+const initialValues = ref(clone(settingsStore.settings.global_search_settings));
+const edits: Ref<Partial<SearchConfig> | null> = ref(null);
+const hasEdits = computed(() => {
+	return edits.value !== null && Object.keys(edits.value).length > 0;
+});
+const saving = ref(false);
+const validationErrors = ref<ValidationError[]>([]);
+
+async function updateSearchSettings(value: SearchConfig) {
+	if (!value) return;
+	validationErrors.value = [];
+	saving.value = true;
+
+	try {
+		const result = SearchConfig.parse(merge({}, initialValues.value, value));
+		await settingsStore.updateSettings({
+			global_search_settings: result,
+		});
+		await settingsStore.hydrate();
+		edits.value = null;
+		saving.value = false;
+		initialValues.value = clone(settingsStore.settings.global_search_settings);
+	} catch (error) {
+		if (error instanceof ZodError) {
+			validationErrors.value = zodErrorToValidationErrors(error, 'global_search_settings');
+		}
+		saving.value = false;
+	}
+}
+
+async function createGlobalSearchSettings() {
+    if(globalSearchSettingsField) return;
+	try {
+		await fieldsStore.createField('directus_settings', {
+			type: 'json',
+			meta: {
+				interface: 'input-code',
+				special: ['cast-json'],
+				display: 'raw',
```

Comment:
We can probably hide this field by default from the user in the project settings